app/ts/Elicitation/PreciseSwingElicitation/PreciseSwingSetWeights/PreciseSwingSetWeights.tsx

```diff
@@ -1,14 +1,39 @@
 import {Grid} from '@material-ui/core';
 import {ElicitationContext} from 'app/ts/Elicitation/ElicitationContext';
 import {PreferencesContext} from 'app/ts/Elicitation/PreferencesContext';
-import React, {useContext} from 'react';
+import _ from 'lodash';
+import React, {useContext, useEffect} from 'react';
+import IExactSwingRatio from '../../Interface/IExactSwingRatio';
 import {getPreciseSwingStatement} from '../PreciseSwingElicitationUtil';
 import OverviewTable from './OverviewTable/OverviewTable';
 
 export default function PreciseSwingSetWeights() {
-  const {mostImportantCriterionId} = useContext(ElicitationContext);
+  const {mostImportantCriterionId, setPreferences} = useContext(
+    ElicitationContext
+  );
   const {criteria} = useContext(PreferencesContext);
 
+  useEffect(initPreferences, [mostImportantCriterionId]);
+
+  function initPreferences() {
+    const preferences: Record<string, IExactSwingRatio> = _(criteria)
+      .filter((criterion) => {
+        return criterion.id !== mostImportantCriterionId;
+      })
+      .map((criterion) => {
+        const preference: IExactSwingRatio = {
+          criteria: [mostImportantCriterionId, criterion.id],
+          elicitationMethod: 'precise',
+          type: 'exact swing',
+          ratio: 1
+        };
+        return [criterion.id, preference];
+      })
+      .fromPairs()
+      .value();
+    setPreferences(preferences);
+  }
```

Comment:
Isn't this something that you want to do in initialization of the context? Also would like to see this tested.

Reply:
Can't initialize beacus we don't know the most important criterion. But will move to util.